edge-modules/ModuleRestarter/src/Settings.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace ModuleRestarter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+
+    [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
+    public class Settings
+    {
+        static readonly Lazy<Settings> DefaultSettings = new Lazy<Settings>(
+            () =>
+            {
+                IConfiguration configuration = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("config/settings.json", optional: true)
+                    .AddEnvironmentVariables()
+                    .Build();
+
+                return new Settings(
+                    configuration.GetValue<string>("ServiceClientConnectionString"),
+                    configuration.GetValue<string>("IOTEDGE_DEVICEID"),
+                    configuration.GetValue<string>("DesiredModulesToRestartCSV"),
+                    configuration.GetValue<int>("RestartIntervalInMins", 10));
+            });
+
+        Settings(
+            string serviceClientConnectionString,
+            string deviceId,
+            string desiredModulesToRestartCSV,
+            int restartIntervalInMins)
+        {
+            this.ServiceClientConnectionString = Preconditions.CheckNonWhiteSpace(serviceClientConnectionString, "ServiceClientConnectionString");
+            this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, "DeviceId");
+            this.RestartIntervalInMins = Preconditions.CheckRange(restartIntervalInMins, 0);
+
+            // mitigate unintended repeated commas
+            this.DesiredModulesToRestart = new List<string>();
+            foreach (string name in desiredModulesToRestartCSV.Split(","))
```

Comment:
You don't need to use foreach here; you can directly call split method with SplitOptions to remove empty entries.  And I think you can return array instead of list.

Reply:
Unfortunately, SplitOptions.RemoveEmptyEntries does not remove whitespace. A solution is outlined [here](https://stackoverflow.com/questions/10682301/stringsplitoptions-removeemptyentries-doesnt-work-as-advertised/10682341) however dotnet 2.0 does not support this.